invenio_search/ext.py

```diff
@@ -406,6 +407,36 @@ def _build(tree_or_filename, alias=None):
                     ignore=ignore,
                 )
 
+    def update_mapping(self, index):
+        """Update mapping of the existing index."""
+        mapping_path = self.mappings[index]
+        index_alias_name = build_alias_name(index)
+
+        # get api returns only dicts
+        index_dict = self.client.indices.get(index_alias_name)
+
+        # make sure only one index exists
+        assert len(list(index_dict.keys())) == 1
+
+        full_index_name = list(index_dict.keys())[0]
+
+        old_mapping = index_dict.keys()[0]["mappings"]
+
+        # need to initialise Index class to use the .put_mapping API wrapper method
+        index_ = dsl.Index(full_index_name, using=self.client)
+
+        with open(mapping_path, "r") as body:
+            mapping = json.load(body)["mappings"]
+            changes = dictdiffer.diff(old_mapping, mapping)
```

Comment:
minor: I'm a bit cautious about verifying the changes in this way. I think it fits our use-cases at the moment, but I could see cases where e.g. we convert a dynamic no-storage object field into a properly typed mapping, in which case I believe it is a valid operation from the ES side, but it would show up as a `modify` operation I the `dictdiffer` output (along with the other `add`s).

Letting ES just fail and print the response error might be good enough, but I'm not sure also how graceful/wanted that is, and also I didn't actually test this locally :)

Reply:
so the risk there is that the field would be deleted from the mapping (AFAIK, ES will not complain about it, because it will be just merged with a new mappings (the PUT works more like a patch in this case, I experienced it in ILS before) and we will never realize that a field was mistakenly removed - I would prefer to validate the update. WDYT?

Reply:
I have the following mapping example in mind:

```python
# this means "store the info in ES but don't index or try to detect types"
{
    "admin_notes": {
        "type": "object",
        "dynamic": false
    }
}
```

which we later change to:

```python
# now we define the mapping properly
{
    "admin_notes": {
        "type": "object",
        "dynamic": "strict",  # this key has been changed
        "properties": {
            "message": {"type": "text"},
            "added_by" {"type": "keyword"}
        }
    }
}
```

Reply:
> I have the following mapping example in mind:
> 
> ```python
> # this means "store the info in ES but don't index or try to detect types"
> {
>     "admin_notes": {
>         "type": "object",
>         "dynamic": false
>     }
> }
> ```
> 
> which we later change to:
> 
> ```python
> # now we define the mapping properly
> {
>     "admin_notes": {
>         "type": "object",
>         "dynamic": "strict",  # this key has been changed
>         "properties": {
>             "message": {"type": "text"},
>             "added_by" {"type": "keyword"}
>         }
>     }
> }
> ```

this will return 400 either way, because in order to do that you need to close the index (always when changing settings, analyzers etc). So it cannot be done by regular PUT to `/_mappings`